src/main/java/com/google/edith/ShelfDataReader.java

```diff
@@ -0,0 +1,103 @@
+package com.google.edith;
+
+import com.google.common.base.Splitter;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Processes data file of product expiration information to populate future user grocery lists. */
+public final class ShelfDataReader {
+  private static final String NO_EXPIRATION = "NO_EXPIRATION";
+
+  /** Finds the specified product in the file. */
+  public String readFile(String itemName) {
+    URL jsonResource = getClass().getClassLoader().getResource("foodkeeper.json");
+    File shelfLifeData = new File(jsonResource.getFile());
+
+    JsonParser jsonParser = new JsonParser();
+    List<JsonArray> potentialMatches = new ArrayList<JsonArray>();
+    try (FileReader reader = new FileReader(shelfLifeData)) {
+      JsonObject data = (JsonObject) jsonParser.parseReader(reader).getAsJsonObject();
+      JsonArray sheets = data.getAsJsonArray("sheets");
```

Comment:
Is it safe to assume sheets is never empty? Or should we add a check?

Reply:
sheets should never be empty with this data file, should I put that in as a comment?

Reply:
Yep I think that's a valid assumption to add - so the reader doesn't have to go to the actual file